File: sharedlife-frontend/src/app/models/task/task.ts
import { TaskTrakingDTO } from "../taskTraking/taskTrakingDTO";

export class Task {
  id: string;
  title: string;
  description: string;
  startDate: string;
  endDate: string;
  finished: boolean;
  user: string;
  idHome: string;
  weekTask: boolean;
  taskTraking: TaskTrakingDTO | null;

  constructor(i?: string, t?: string, d?: string, sd?: string, ed?: string, f?:boolean, u?:string, idHome?:string, weekTask?:boolean, tk?: TaskTrakingDTO){
    this.id = i ?? "";
    this.title = t ?? "";
    this.description = d ?? "";
    this.startDate = sd ?? "";
    this.endDate = ed ?? "";
    this.finished = f ?? false;
    this.user = u ?? "";
    this.idHome = idHome ?? "";
    this.weekTask = weekTask ?? false;
    this.taskTraking = tk ?? null;
  }
}

File: sharedlife-backend/src/main/java/net/tfg/sharedlife/model/Task.java
package net.tfg.sharedlife.model;

import java.util.Date;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import lombok.Data;
import lombok.EqualsAndHashCode;
import net.tfg.sharedlife.enums.HomeRoomEnum;

@Data
@EqualsAndHashCode
@Entity
@Table(name = "tasks")
public class Task {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	// Titulo de la tarea
	@Column(name = "title")
	private String title;
	
	// Descripción de la tarea
	@Column(name = "description")
	private String description;
	
	//Fecha de inicio de la tarea
	@Column(name = "startDate")
	private Date startDate;
	
	//Fecha de fin de tarea
	@Column(name = "endDate")
	private Date endDate;
	
	// Indica si la tarea esta terminada o no
	@Column(name = "finished")
	private boolean finished;

	// Indica si la tarea es una tarea semana o no
	@Column(name = "weekTask")
	private Boolean weekTask;

	// Enumerado que indica si la tarea pertenece a una habitacion de la casa
	@Column(name = "homeRoom")
	private HomeRoomEnum homeRoom;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "id_user")
	private User user; // RESPONSABLE DE LA TAREA
	
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "id_home")
	private Home home;

	@OneToOne(cascade = CascadeType.ALL)
	@JoinColumn(name = "id_taskTraking")
	private TaskTraking taskTraking; 
}
